crates/ruff_cli/src/diagnostics.rs

```diff
@@ -177,7 +177,23 @@ impl AddAssign for FixMap {
     }
 }
 
+fn get_override_source_type(
+    path: Option<&Path>,
+    extension: &FxHashMap<String, Language>,
+) -> Option<PySourceType> {
+    let Some(ext) = path.and_then(|p| p.extension()).and_then(|p| p.to_str()) else {
+        return None;
+    };
+    match extension.get(ext) {
+        Some(Language::Python) => Some(PySourceType::Python),
```

Comment:
Sorry for not noticing this in my first review 😓 

How will this work if the cell contained IPython escape commands (`!pwd`, `%matplotlib inline`)? The parser mode is tied up with `PySourceType`. This mode is then used to allow the escape commands.

https://github.com/astral-sh/ruff/blob/2d5ce4532ade11710ff854516cd085b9ea629cf2/crates/ruff_python_parser/src/lib.rs#L325-L332

Now, if the cell contained escape commands and the override flag tells that this is a Python language (`--extension-override ipynb:python`), then it'll throw a syntax error.

Reply:
`jupyterlab-lsp` can  replace notebook magics with equivalent python code in most cases, so in this use case these syntax errors are avoided (I think the code is [here](https://github.com/jupyter-lsp/jupyterlab-lsp/blob/5a8686067bdeeb1f8ce5bc108d7fd047809b0f80/packages/jupyterlab-lsp/src/transclusions/ipython/overrides.ts#L58)).

Any python code affected by magic ends up in a string so doesn't get checked, for example with magics like `%time` and `%%time`.

Reply:
Thanks for letting us know about that.

Coming from a _public_ API perspective, this might still be a problem as, for example, if someone's trying to use it in a similar manner with an integration that doesn't perform those transformations, we'd throw an error.

I would love any other opinion on this. I'm fine with including the flag but if we include it in the config, then it's part of the public API which is where I'm unsure of.

Reply:
FWIW `jupyter nbconvert --to script` also seems to convert notebook magics to python in the same way. [`nbdev`](https://github.com/fastai/nbdev) has a routine callled `scrub_magics` which removes magics altogether, though it seems to be optional.

The docs could make it clear that `python` means `python` without notebookisms. Maybe something like:

> Users who use this option to lint code extracted from notebooks as python should ensure that the code should be free of IPython magics, as these will cause a syntax error.